agent/mapcall/src/main/java/randoop/instrument/ReplacementFileReader.java

```diff
@@ -0,0 +1,557 @@
+        continue;
+      }
+      File file = new File(pathString);
+      if (!file.exists()) {
+        continue;
+      }
+      if (file.isDirectory()) {
+        Path path = file.toPath();
+        Path replacementPath = path.resolve(replacementPackage.replace('.', File.separatorChar));
+        if (Files.exists(replacementPath) && Files.isDirectory(replacementPath)) {
+          addPackageReplacements(
+              replacementMap, originalPackage, replacementPackage, replacementPath.toFile());
+          found = true; // directory for package was found
+        }
+      } else { // or a jar file
+        try {
+          JarFile jarFile = new JarFile(file);
+          if (addPackageReplacements(
+              replacementMap, originalPackage, replacementPackage, jarFile)) {
+            found = true;
+          }
+        } catch (IOException e) {
+          throw new ReplacementException(
+              "Error reading jar file from boot classpath: " + file.getName());
+        }
+      }
+    }
+    if (!found) {
+      String msg =
+          String.format(
+              "No package or class for replacement %s found on the boot classpath",
+              replacementPackage);
+      throw new ReplacementException(msg);
+    }
+  }
+
+  /**
+   * Discovers and adds method replacements from the given {@code ClassLoader} for the package
+   * replacement determined by the {@code originalPackage} and {@code replacementPackage} strings.
+   * Contrasts with {@link #discoverPackageReplacements(ConcurrentHashMap, String, String)} that
+   * searches for the package on the boot classpath.
+   *
+   * @see #discoverClassOrPackageReplacements(ConcurrentHashMap, String, String)
+   * @param replacementMap the method replacement map to which new replacements are added
+   * @param originalPackage the original package name
+   * @param replacementPackage the replacement package name
+   * @param loader the {@code ClassLoader}
+   * @throws ReplacementException if no package corresponding to replacement is found
+   */
+  private static void discoverPackageReplacements(
+      ConcurrentHashMap<MethodSignature, MethodSignature> replacementMap,
+      String originalPackage,
+      String replacementPackage,
+      ClassLoader loader)
+      throws ReplacementException {
+    boolean found = false;
+    Enumeration<URL> resources;
+    try {
+      resources = loader.getResources(replacementPackage.replace('.', '/'));
+    } catch (IOException e) {
+      throw new ReplacementException(e.getMessage());
+    }
+    while (resources.hasMoreElements()) {
+      URL url = resources.nextElement();
+      try {
+        URLConnection connection = url.openConnection();
+        if (connection instanceof JarURLConnection) {
+          JarFile jarFile = ((JarURLConnection) connection).getJarFile();
+          if (addPackageReplacements(
+              replacementMap, originalPackage, replacementPackage, jarFile)) {
+            found = true;
```

Comment:
All replacements are stored internally in terms of methods, right?  (I believe that packages are just a user interface mechanism for specifying multiple methods.)

So the only reason to continue processing is to find a different method replacement.  That's what my comment is about; I'm asking why that is desirable.

Or do you expect that users will supply multiple different package replacements?  The user manual doesn't seem to say anything about that.

I'm confused about the feature, its purpose, and how it would be used.